### PR TITLE
Log error for invalid beatmap/storyboard events instead of throwing

### LINKED ISSUE #5526: InvalidDataException: Unknown event type
Sentry Issue: [OSU-LAZER-46](https://sentry.io/organizations/ppy/issues/652084602/?referrer=github_integration)

```
InvalidDataException: Unknown event type 
  Module "osu.Game.Beatmaps.Formats.LegacyBeatmapDecoder", line 48, in handleEvent
    Void handleEvent(System.String)
  Module "osu.Game.Beatmaps.Formats.LegacyDecoder`1", line 98, in ParseStreamInto
    Void ParseStreamInto(System.IO.StreamReader, T)

Unknown event type 
```

At *very* least, should not be sent to sentry, not should fail a beatmap import.

### PR DESCRIPTION
**Pull request by nyquillerium**
Resolves https://github.com/ppy/osu/issues/5526

Also adds a test to check a beatmap with a single bad event is parsed fine.

I copied the severity/target from https://github.com/nyquillerium/osu/blob/cd6fe918821aec9d150311bcc6ac0ec653b672dc/osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs#L404

### CODE REVIEW

**Comment by peppy** [comment]
The test .osu file can probably be 10 lines or less.

File: osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs [hunk 1]
```diff
@@ -290,8 +290,12 @@ private void handleEvent(string line)
             string[] split = line.Split(',');
 
             EventType type;
+
             if (!Enum.TryParse(split[0], out type))
-                throw new InvalidDataException($@"Unknown event type {split[0]}");
+            {
+                Logger.Log($"A beatmap event could not be parsed and will be ignored: {split[0]}", LoggingTarget.Runtime, LogLevel.Important);
```
Comment (peppy) [inline osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs]: why is it not mentioning type?

File: osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs [hunk 2]
```diff
@@ -290,8 +290,12 @@ private void handleEvent(string line)
             string[] split = line.Split(',');
 
             EventType type;
+
             if (!Enum.TryParse(split[0], out type))
-                throw new InvalidDataException($@"Unknown event type {split[0]}");
+            {
+                Logger.Log($"A beatmap {type} event could not be parsed and will be ignored: {split[0]}", LoggingTarget.Runtime, LogLevel.Important);
```
Comment (peppy) [inline osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs]: you did not test this.

File: osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs]: please read the original exception. please add a message matching it. it should mention that `{split[0]}` is the **type**.

File: osu.Game.Tests/Beatmaps/Formats/LegacyBeatmapDecoderTest.cs [hunk 3]
```diff
@@ -482,5 +482,23 @@ public void TestDecodeHitObjectNullAdditionBank()
                 Assert.AreEqual(hitObjects[0].Samples[0].Bank, hitObjects[0].Samples[1].Bank);
             }
         }
+
+        [Test]
+        public void TestDecodeInvalidEvents()
```
Comment (peppy) [inline osu.Game.Tests/Beatmaps/Formats/LegacyBeatmapDecoderTest.cs]: this test is very nondescript. someone looking at this is going to WTF. there's no asserts on events of any kind, so what is it even doing?

Comment (nyquillerium) [reply osu.Game.Tests/Beatmaps/Formats/LegacyBeatmapDecoderTest.cs]: You can kind of argue the same for the rest of the tests in this file though? I agree the name doesn't really help much for the cause though. Would it be alright if I renamed + xmldocd this test?

File: osu.Game/Beatmaps/Formats/LegacyDecoder.cs [hunk 4]
```diff
@@ -43,14 +43,7 @@ protected override void ParseStreamInto(StreamReader stream, T output)
                     continue;
                 }
 
-                try
-                {
-                    ParseLine(output, section, line);
-                }
-                catch (Exception e)
-                {
-                    Logger.Error(e, $"Failed to process line \"{line}\" into {output}");
```
Comment (peppy) [inline osu.Game/Beatmaps/Formats/LegacyDecoder.cs]: you are hiding information here. this used to tell you which line the failure happened on; now it doesn't.

Comment (nyquillerium) [reply osu.Game/Beatmaps/Formats/LegacyDecoder.cs]: The new commit will show the message as it was before:
```
[runtime:important] 2019-08-07 10:48:53: Failed to process line "bad,event,this,should,fail" into osu.Game.Beatmaps.Beatmap: This is an exception
[runtime:important] 2019-08-07 10:48:53: Failed to process line "this,is,also,bad" into osu.Game.Beatmaps.Beatmap: This is an exception
```

**Review by smoogipoo — DISMISSED** [review]

File: osu.Game/Beatmaps/Formats/LegacyStoryboardDecoder.cs [hunk 5]
```diff
@@ -82,8 +83,12 @@ private void handleEvents(string line)
                 storyboardSprite = null;
 
                 EventType type;
+
                 if (!Enum.TryParse(split[0], out type))
-                    throw new InvalidDataException($@"Unknown event type {split[0]}");
+                {
+                    Logger.Log($"Unknown storyboard event of type {split[0]} could not be parsed and will be ignored.", LoggingTarget.Runtime, LogLevel.Important);
```
Comment (peppy) [inline osu.Game/Beatmaps/Formats/LegacyStoryboardDecoder.cs]: this can be reverted

File: osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs [hunk 6]
```diff
@@ -290,8 +290,12 @@ private void handleEvent(string line)
             string[] split = line.Split(',');
 
             EventType type;
+
             if (!Enum.TryParse(split[0], out type))
-                throw new InvalidDataException($@"Unknown event type {split[0]}");
+            {
+                Logger.Log($"Unknown beatmap event of type {split[0]} could not be parsed and will be ignored.", LoggingTarget.Runtime, LogLevel.Important);
```
Comment (peppy) [inline osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs]: this can be reverted

**Review by peppy — CHANGES_REQUESTED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**